### PR TITLE
add symbol for '__printf_chk'

### PR DESCRIPTION
**Pull request by naoki9911**

### CODE REVIEW

**Comment by Saza-ku** [comment]
I have never been asked for this symbol.

@ainozaki You also mentioned the problem. Do you know something about it?

**Comment by naoki9911** [comment]
I encountered this issue when I simply cloned this repository and built with `zig build`.

I got below error messages.
```
zig build-exe mewz.elf Debug x86_64-freestanding: error: the following command failed with 1 compilation errors:
/home/naoki/.local/zig-master/zig build-exe /tmp/mewz/src/main.zig /tmp/mewz/src/boot.S /tmp/mewz/src/interrupt.S /tmp/mewz/build/newlib/libc.a /tmp/mewz/build/lwip/libtcpip.a /tmp/mewz/build/lwip/liblwipcore.a /tmp/mewz/build/lwip/liblwipallapps.a -cflags -I submodules/newlib/newlib/libc/include -- /tmp/mewz/src/c/newlib_support.c -cflags -I submodules/newlib/newlib/libc/include -- /tmp/mewz/src/c/lwip_support.c --cache-dir /tmp/mewz/zig-cache --global-cache-dir /home/naoki/.cache/zig --name mewz.elf -static -mcmodel kernel -target x86_64-freestanding -mcpu x86_64-mmx+soft_float-sse-sse2 --script /tmp/mewz/src/x64.ld --mod options::/tmp/mewz/zig-cache/c/2e24cb6c7e49d570a5020a1af9df5ef9/options.zig --deps options --listen=- 
Build Summary: 3/6 steps succeeded; 1 failed (disable with --summary none)
install transitive failure
└─ install mewz.elf transitive failure
   └─ zig build-exe mewz.elf Debug x86_64-freestanding 1 errors
error: ld.lld: undefined symbol: __printf_chk
    note: referenced by tcpip.c
    note:               tcpip.o:(accept_callback) in archive /tmp/mewz/build/lwip/libtcpip.a
    note: referenced by ethernet.c
    note:               ethernet.o:(ethernet_output) in archive /tmp/mewz/build/lwip/liblwipcore.a
    note: referenced by pbuf.c
    note:               pbuf.o:(pbuf_remove_header.part.0) in archive /tmp/mewz/build/lwip/liblwipcore.a
    note: referenced 244 more times
```

**Comment by ainozaki** [comment]
Yes, I encountered this error, too.
TL;DR, this commit will resolve the problem.
https://github.com/mewz-project/mewz/commit/ba4bf410905ff3717fc6a4d3ac9cadf4c4df02f9

**Comment by ainozaki** [comment]
To supplement, the issue arises because `printf` used in lwip is replaced with `printf_chk`, which includes stack checks. However, since newlib only has `printf`, linking fails. It will be fixed by setting a flag to prevent `printf` from being replaced.

**Comment by Saza-ku** [comment]
@ainozaki So should we handle the problem with https://github.com/mewz-project/mewz/commit/ba4bf410905ff3717fc6a4d3ac9cadf4c4df02f9?

Or would it be better to merge this PR?

**Comment by ainozaki** [comment]
> @ainozaki So should we handle the problem with https://github.com/mewz-project/mewz/commit/ba4bf410905ff3717fc6a4d3ac9cadf4c4df02f9?
> 
> 
> 
> Or would it be better to merge this PR?

I'm not sure if this problem will always occur, I'm good with handling by the commit without merging it.

**Comment by Saza-ku** [comment]
So we must merge https://github.com/mewz-project/mewz/commit/ba4bf410905ff3717fc6a4d3ac9cadf4c4df02f9, right?

**Comment by Saza-ku** [comment]
@naoki9911 we just merged #50. Does this fix the issue?

**Comment by naoki9911** [comment]
Fixed in my environment too. Thanks!

**Comment by Saza-ku** [comment]
Okay. So we'll close this PR. Thanks for reporting the issue!
